app/javascript/components/Greeting.js

```diff
@@ -0,0 +1,27 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+
+const Greeting = () => {
+  const [message , getMessage] = useState(["hi"]);
+  //const message = useSelector((state) => state.greeting);
+
+  async function fetchMessage() {
+    await axios.get('api/messages').then((response) => {
+      getMessage(response.data.greeting);
+      console.log({message})
+    });
+  }
+
+  // useEffect(() => {
+  //   fetchMessage();
+  // }, []);
+
+  return (
+    <div>
+      <h1>{message}</h1>
+      <button onClick={() => fetchMessage()}>Greeting message</button>
+    </div>
+  );
+};
+
+export default Greeting;
```

Comment:
- [ ] Please make sure that your app has a store, an action, and a reducer in your redux folder using the duck pattern instead of fetching the messages in the `Greeting` component, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/rails_react_webpack.md#:~:text=Create%20a%20store%2C%20an%20action%2C%20and%20a%20reducer%20that%20will%20connect%20to%20your%20API%20endpoint%20to%20get%20the%20random%20greeting.) requirement.

- [ ] Kindly remove any unwanted or commented code you might not want so that your project appears more professional.